### PR TITLE
Fix nil point panic when store failed.

### PR DESCRIPTION
**Pull request by shenli**
When running session.GetTxn(), we should check the error returned by isAutocommit().
@coocood @zimulala @tiancaiamao PTAL

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

File: tidb.go [hunk 1]
```diff
@@ -150,7 +150,11 @@ func runStmt(ctx context.Context, s ast.Statement, args ...interface{}) (ast.Rec
 	se := ctx.(*session)
 	se.history.add(0, s)
 	// MySQL DDL should be auto-commit.
-	if s.IsDDL() || autocommit.ShouldAutocommit(ctx) {
+	ac, err1 := autocommit.ShouldAutocommit(ctx)
```
Comment (zimulala) [inline tidb.go]: Can use `err` here？

Comment (shenli) [inline tidb.go]: No we should keep the error and check if we should commit or rollback.

File: session.go [hunk 2]
```diff
@@ -551,7 +557,11 @@ func (s *session) GetTxn(forceNew bool) (kv.Transaction, error) {
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		if !s.isAutocommit(s) {
+		ac, err1 := s.isAutocommit(s)
```
Comment (qiuyesuifeng) [inline session.go]: Why not using err here?

Comment (shenli) [inline session.go]: It will override err.
err is used in Line 574.

**Comment by coocood** [comment]
LGTM

File: session.go (hunk 2 above)
Comment (shenli) [inline session.go]: PTAL @qiuyesuifeng

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**